tests/platform_tests/counterpoll/test_counterpoll_watermark.py

```diff
@@ -0,0 +1,233 @@
+    3. run one of the (type[watermark,queue,pg-drop]) (randomized)
+        a. queue: enable the queue counter polling, check whether queue map is generated in COUNTERS_DB,
+           no WATERMARK or PG-DROP stats in FLEX_COUNTER_DB
+        b. watermark: enable the watermark counter polling, check whether PG and queue maps are generated in COUNTER_DB
+           and both PG and QUEUE WATERMARK stats exists in FLEX_COUNTER_DB, without any QUEUE or PG-DROP stats
+           for watermark also check WATERMARK stats in COUNTERS_DB (USER/PERSISTENT/PERIODIC WATERMARKS)
+        c. pg-drop: enable the pg-drop counter polling, check whether PG map is generated in COUNTERS_DB
+           no WATERMARK or QUEUE stats in FLEX_COUNTER_DB
+    4. enables all three counterpolls (queue,watermark,pg-drop) and count stats per type
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    skip_release(duthost, ["202205", "202111", "202106", "202012", "201911", "201811", "201803"])
+
+    counted_dict = {}
+    # choosing only one between reload / reboot due to test duration limitations
+    with allure.step("choosing random config apply method"):
+        config_apply_method = random.choice(["config reload", "switch reboot"])
+    with allure.step("disabling all counterpolls"):
+        ConterpollHelper.disable_counterpoll(duthost, CounterpollConstants.COUNTERPOLL_MAPPING.values())
+
+    # verify relevant counterpolls (queue/watermark/pg-drop) are disabled
+    with allure.step("Verifying initial output of {} on {} ..."
+                     .format(CounterpollConstants.COUNTERPOLL_SHOW, duthost.hostname)):
+        verify_all_counterpoll_status(duthost, DISABLE)
+
+    with allure.step("saving config on dut {} after counterpoll disable...".format(duthost.hostname)):
+        duthost.shell('config save -y')
+
+    # choosing only one counterpoll to test due to test duration limitaions
+    tested_counterpoll = random.choice(RELEVANT_COUNTERPOLLS)
+    # need reload or reboot after disabling counterpolls to clean DB stats, this is by design
+    with allure.step(config_apply_method + " dut {} ...".format(duthost.hostname)):
+        if 'reload' in config_apply_method:
+            config_reload(duthost)
+        elif 'reboot' in config_apply_method:
+            reboot(duthost, localhost)
+    # verify all counterpolls are disabled after reload or reboot
+    with allure.step("Verifying output of {} on {} after {} ..."
+                     .format(CounterpollConstants.COUNTERPOLL_SHOW, duthost.hostname, config_apply_method)):
+        verify_all_counterpoll_status(duthost, DISABLE)
+    # enable the selected counterpoll queue/watermark/pg-drop
+    with allure.step("enabling randomly selected counterpoll {} on {} ..."
+                     .format(duthost.hostname, [tested_counterpoll])):
+        ConterpollHelper.enable_counterpoll(duthost, [tested_counterpoll])
+    verify_counterpoll_status(duthost, [tested_counterpoll], ENABLE)
```

Comment:
Align this line into the with block?

Reply:
verify_counterpoll_status isn't part of the enablement but can be aligned under this with block and change the comment printed accordingly

Reply:
Done